pkg/controller/multiclusterstatusaggregation/multiclusterStatusAggregation_controller.go

```diff
@@ -0,0 +1,618 @@
+			HealthStatus: appSetClusterStatusMap[appset][cluster].HealthStatus,
+		}
+
+		// Calculate the summary while we're here.
+		clusters++
+
+		switch appSetClusterStatusMap[appset][cluster].HealthStatus {
+		case "Healthy":
+			healthy++
+		case "Progressing":
+			inProgress++
+			notHealthy++
+		default:
+			notHealthy++
+		}
+
+		switch appSetClusterStatusMap[appset][cluster].SyncStatus {
+		case "Synced":
+			synced++
+		default:
+			notSynced++
+		}
+	}
+
+	summary := appsetreportV1alpha1.ReportSummary{
+		Synced:     strconv.Itoa(synced),
+		NotSynced:  strconv.Itoa(notSynced),
+		Healthy:    strconv.Itoa(healthy),
+		NotHealthy: strconv.Itoa(notHealthy),
+		InProgress: strconv.Itoa(inProgress),
+		Clusters:   strconv.Itoa(clusters),
+	}
+	if clusters != (synced+notSynced) || clusters != (healthy+notHealthy) {
+		klog.Warningf("Total number of clusters does not add up, %v", summary)
+	}
+
+	// Combine cluster conditions from manifestwork and yaml
+	for _, item := range appSetCRDConditions {
+		klog.V(4).Info("AppSetCRD conditions item: ", item)
+
+		if e, ok := appSetClusterConditionsMap[item.Cluster]; ok {
+			e.Conditions = item.Conditions
+			appSetClusterConditionsMap[item.Cluster] = e
+		} else {
+			appSetClusterConditionsMap[item.Cluster] = item
+		}
+	}
+
+	res := make([]appsetreportV1alpha1.ClusterCondition, 0, len(appSetClusterConditionsMap))
+
+	for _, condition := range appSetClusterConditionsMap {
+		res = append(res, condition)
+	}
+
+	// Sort after list has been created so it's in a natural order
+	sort.Sort(AppSetClusterConditionsSorter(res))
+	klog.V(4).Info("Sorted cluster conditions list", res)
+
+	return summary, res
+}
+
+func (r *ReconcilePullModelAggregation) newAppSetReport(appsetNs, appsetName string, appsetResources []appsetreportV1alpha1.ResourceRef,
+	appsetClusterConditions []appsetreportV1alpha1.ClusterCondition,
+	appsetSummary appsetreportV1alpha1.ReportSummary) *appsetreportV1alpha1.MulticlusterApplicationSetReport {
+	newAppSetReport := &appsetreportV1alpha1.MulticlusterApplicationSetReport{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "MulticlusterApplicationSetReport",
+			APIVersion: "apps.open-cluster-management.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appsetName,
+			Namespace: appsetNs,
+			Labels: map[string]string{
+				"apps.open-cluster-management.io/hosting-applicationset": fmt.Sprintf("%.63s", appsetNs+"."+appsetName),
+			},
+		},
+		Statuses: appsetreportV1alpha1.AppConditions{
+			Resources:         appsetResources,
+			ClusterConditions: appsetClusterConditions,
+			Summary:           appsetSummary,
+		},
+	}
+
+	return newAppSetReport
+}
+
+func (r *ReconcilePullModelAggregation) compareAppSetReports(report1, report2 *appsetreportV1alpha1.MulticlusterApplicationSetReport) bool {
+	isSame := true
+
+	if !equality.Semantic.DeepEqual(report1.GetLabels(), report2.GetLabels()) {
+		klog.V(4).Info("Labels not same")
+
+		isSame = false
```

Comment:
@lennysgarage  it is better to use `klog.Info("Labels not same")` in the function. We will know which section is not the same when comparing appsetReport spec.  This would be very helpful info in troubleshooting

Reply:
Updated!